pkg/apis/extensions/register.go

```diff
@@ -52,8 +52,6 @@ func addKnownTypes(scheme *runtime.Scheme) {
 		&DeploymentRollback{},
 		&HorizontalPodAutoscaler{},
 		&HorizontalPodAutoscalerList{},
-		&Job{},
```

Comment:
Change to batch.Job & batch.JobList to fix the error you're seeing. (You'll have to import the batch package.) You can't get rid of this completely because they still exist in extensions v1beta1.